src/utils/fonts/index.js

```diff
@@ -33,12 +34,8 @@ export const normalize = (size) => {
   return normalizedSize;
 };
 
-export const normalizeFontSize = (fontSize) => {
-  const referenceHeight = 771;
-  const currentScreenHeight = Dimensions.get('screen').height;
-  const normalizedFontSize = (currentScreenHeight / referenceHeight) * fontSize;
-  return normalizedFontSize;
-};
+export const normalizeFontSize = (size, factor = 0.5) =>
+  size + (dimen.normalizeDimen(size) - size) * factor;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of `normalizeFontSize` introduces a dependency on the `dimen.normalizeDimen` function. This change could potentially introduce bugs if the `dimen.normalizeDimen` function is not thoroughly tested or if it does not handle edge cases properly. It's recommended to ensure that `dimen.normalizeDimen` has comprehensive test coverage and handles edge cases appropriately.

Additionally, the introduction of the `factor` parameter without any boundary checks can lead to unexpected results if an inappropriate value is passed. Consider adding checks to ensure that `factor` is within a reasonable range (e.g., 0 to 1).

```diff
- export const normalizeFontSize = (size, factor = 0.5) =>
-   size + (dimen.normalizeDimen(size) - size) * factor;
+ export const normalizeFontSize = (size, factor = 0.5) => {
+   if (factor < 0 || factor > 1) {
+     throw new Error('Factor must be between 0 and 1');
+   }
+   return size + (dimen.normalizeDimen(size) - size) * factor;
+ }
```
This will help prevent potential issues caused by incorrect usage of the `factor` parameter.


<!-- This is an auto-generated comment by OSS CodeRabbit -->